src/compiler/checker.ts

```diff
@@ -8060,6 +8118,65 @@ namespace ts {
             return links.resolvedType;
         }
 
+        function getActualTypeParameter(type: Type) {
+            return type.flags & TypeFlags.Substitution ? (<SubstitutionType>type).typeParameter : type;
+        }
+
+        function createConditionalType(checkType: Type, extendsType: Type, trueType: Type, falseType: Type, target: ConditionalType, mapper: TypeMapper, aliasSymbol: Symbol, aliasTypeArguments: Type[]) {
+            const type = <ConditionalType>createType(TypeFlags.Conditional);
+            type.checkType = checkType;
+            type.extendsType = extendsType;
+            type.trueType = trueType;
+            type.falseType = falseType;
+            type.target = target;
+            type.mapper = mapper;
+            type.aliasSymbol = aliasSymbol;
+            type.aliasTypeArguments = aliasTypeArguments;
+            return type;
+        }
+
+        function getConditionalType(checkType: Type, extendsType: Type, baseTrueType: Type, baseFalseType: Type, target: ConditionalType, mapper: TypeMapper, aliasSymbol?: Symbol, baseAliasTypeArguments?: Type[]): Type {
+            // Return union of trueType and falseType for any and never since they match anything
+            if (checkType.flags & (TypeFlags.Any | TypeFlags.Never)) {
+                return getUnionType([instantiateType(baseTrueType, mapper), instantiateType(baseFalseType, mapper)]);
+            }
+            // Return trueType for a definitely true extends check
+            if (isTypeAssignableTo(checkType, extendsType)) {
+                return instantiateType(baseTrueType, mapper);
+            }
+            // Return falseType for a definitely false extends check
+            if (!isTypeAssignableTo(instantiateType(checkType, anyMapper), instantiateType(extendsType, constraintMapper))) {

```

Comment:
Could you document why we use an `anyMapper` here instead of a `neverMapper`, and what the possible difference would be? IMO, the difference isn't obvious.